auraed/src/runtime/cell_service/cells/cgroups/cgroup.rs

```diff
@@ -42,26 +46,57 @@ pub struct Cgroup {
 
 impl Cgroup {
     pub fn new(cell_name: CellName, spec: CgroupSpec) -> Self {
-        let CgroupSpec { cpu_cpus, cpu_quota, cpu_weight, cpuset_mems } = spec;
+        let CgroupSpec { cpu, cpuset } = spec;
 
         // NOTE: v2 cgroups can either have nested cgroups or processes, not both (leaf workaround)
         // NOTE: '_' is a disallowed character in cell name, so won't collide
         let name = format!("{cell_name}/_");
-        let hierarchy = hierarchy();
-
-        let builder = CgroupBuilder::new(&name)
-            // CPU Controller
-            .cpu()
-            .shares(cpu_weight.into_inner())
-            .mems(cpuset_mems.into_inner())
-            .period(1000000) // microseconds in a second
-            .quota(cpu_quota.into_inner())
-            .cpus(cpu_cpus.into_inner())
-            .done()
-            // Final Build
-            .build(hierarchy);
-
-        let inner = builder.expect("inner from builder");
+
+        let builder = CgroupBuilder::new(&name);
+
+        // cpu controller
+        let builder = if let Some(CpuController { weight, max }) = cpu {
+            let builder = builder.cpu();
+
+            let builder = if let Some(weight) = weight {
+                builder.shares(weight.into_inner())
+            } else {
+                builder
+            };
+
+            let builder = if let Some(max) = max {
+                builder.quota(max.into_inner()).period(1_000_000) // microseconds in a second
```

Comment:
note i have a PR creating a const for this with docs.  feel free to absorb it or merge it.

Reply:
already merged and rebasing on it